tools/itsg33-to-gh-issues/tests/test_script.py

```diff
@@ -0,0 +1,81 @@
+from unittest.mock import patch,Mock
+import pytest
+import script
+import secrets
+
+
+@patch("script.get_controls")
+@patch("script.post_request")
+def test_main(mock_post_request, mock_get_controls):
+    mock_get_controls.return_value = [
+        ["Family1", "Control1", "E1", "Control1 name", "Class1", "Definition1", "Supplemental guidance1", "References1", "IT Security Function", "", "", "", "", "", "Medium", "", "", ""],
+        ["Family2", "Control2", "E2", "Control2 name", "Class2", "Definition2", "Supplemental guidance2", "References2", "", "IT Operation Group", "", "", "", "", "Low", "", "", ""]
+    ]
+    mock_post_request.return_value.status_code = 201
+
+    script.main()
+
+    assert mock_post_request.call_count == 2
+
+@patch("script.requests.post")
+def test_post_request_failure():
+    issues_url = "https://api.github.com/repos/owner/repo/issues"
+    headers = {'Authorization': 'Bearer token'}
+    issues_json = {"title": "Test Issue", "body": "This is a test issue"}
+    response_mock = Mock(status_code=400, text="Bad request")
+    
+    with patch('requests.post', return_value=response_mock):
+        response = script.post_request(issues_url, headers, issues_json)
+        assert response == response_mock
+        assert "Failed to create issue for control" in pytest.capsys.readouterr().out
+        assert "Response: Bad request" in pytest.readouterr().out
```

Comment:
I think you could get rid of the `with` block and do it like so since you're already patching with the annotation:
```suggestion
@patch("script.requests.post")
def test_post_request_failure(mock_post):
    issues_url = "https://api.github.com/repos/owner/repo/issues"
    headers = {'Authorization': 'Bearer token'}
    issues_json = {"title": "Test Issue", "body": "This is a test issue"}
    response_mock = Mock(status_code=400, text="Bad request")
    mock_post.return_value = response_mock
    
    response = script.post_request(issues_url, headers, issues_json)
    assert response == response_mock
    assert "Failed to create issue for control" in pytest.capsys.readouterr().out
    assert "Response: Bad request" in pytest.readouterr().out
```

Reply:
Muchas gracias!